graph-commons/src/main/scala/io/renku/data/Message.scala

```diff
@@ -66,6 +69,12 @@ object Message extends MessageCodecs {
           .map(toSingleLine)
           .getOrElse(throw new IllegalArgumentException("Message cannot be instantiated with a blank String"))
       }
+
+    def fromJsonUnsafe(message: Json): Message =
+      if (message.isNull || message == Json.obj())
```

Comment:
Not sure, an empty array could also occur? If you go further, there could be arrays of empty objects 🤯 😄 But I think we can assume some kind of healthy mind using this, I hope 😄 And if you obtain it from some api, then you'll probably want to see even empty data structures?

Reply:
Yes, I think it's good to make it more robust. I added some additional checks.